Mathlib/Data/ZMod/Factorial.lean

```diff
@@ -0,0 +1,42 @@
+/-
+Copyright (c) 2023 Moritz Firsching. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Moritz Firsching
+-/
+import Mathlib.Data.Nat.Factorial.BigOperators
+import Mathlib.Data.ZMod.Basic
+
+/-!
+# Facts about factorials in ZMod
+
+We collect facts about factorials in context of modular arithmetic.
+
+## Main statements
+
+* `ZMod.cast_descFactorial`: For natural numbers `n` and `p`, where `n` is less than or equal to `p`
+  the descending factorial of `(p - 1)` taken `n` times modulo `p` equals `(-1) ^ n * n!`.
+
+## See also
+
+For the prime case and involving `factorial` rather than `descFactorial`, see Wilson's theorem:
+* Nat.prime_iff_fac_equiv_neg_one
+
+-/
+
+open Finset Nat
+
+namespace ZMod
+
+theorem cast_descFactorial {n p : ℕ} (h : n ≤ p) :
```

Comment:
Can you use this to golf the proof of Wilson's lemma? https://github.com/leanprover-community/mathlib4/blob/1c9929f3360c26cc93d43c44771f062fec6528fc//Mathlib/NumberTheory/Wilson.lean#L40

Reply:
No, because `(p - 1)! = descFactorial (p - 1) (p - 1)` and all the current lemma is saying is `descFactorial (p - 1) (p - 1) = (-1)^(p - 1) * (p - 1)! = (p - 1)!`!

Reply:
I don't think so. As explained above then setting `n := p - 1` we get a tautology and not Wilson's lemma.

Reply:
ok, of course! sorry for the noise